include/ddc/policy.hpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+namespace ddc {
+
+/// Serial execution on the host
+struct serial_host_policy
+{
+};
+
+
+/// Parallel execution on the default device
+struct parallel_host_policy
+{
+};
+
+/// Kokkos parallel execution uisng MDRange policy
+struct parallel_device_policy
+{
+};
+
+using default_policy = serial_host_policy;
+
+namespace policies {
+
+inline constexpr serial_host_policy serial_host;
+inline constexpr parallel_host_policy parallel_host;
+inline constexpr parallel_device_policy parallel_device;
+
+constexpr serial_host_policy const& policy(::Kokkos::Serial const&)
+{
+    return serial_host;
+}
+
+#ifdef KOKKOS_ENABLE_CUDA
+constexpr parallel_device_policy const& policy(::Kokkos::Cuda)
+{
+    return parallel_device;
+}
+#endif
+
+#ifdef KOKKOS_ENABLE_HIP
+constexpr parallel_device_policy const& policy(::Kokkos::HIP)
+{
+    return parallel_device;
+}
+#endif
+
+#ifdef KOKKOS_ENABLE_OPENMP
+constexpr parallel_host_policy const& policy(::Kokkos::OpenMP)
+{
+    return parallel_host;
+}
+#endif
```

Comment:
The Kokkos execution spaces do not seem to be `constexpr` compatible, thus the error in the CI. We did not see that before because the `policy` function used to be template.

```suggestion
serial_host_policy const& policy(Kokkos::Serial)
{
    return serial_host;
}

#ifdef KOKKOS_ENABLE_CUDA
parallel_device_policy const& policy(Kokkos::Cuda)
{
    return parallel_device;
}
#endif

#ifdef KOKKOS_ENABLE_HIP
parallel_device_policy const& policy(Kokkos::HIP)
{
    return parallel_device;
}
#endif

#ifdef KOKKOS_ENABLE_OPENMP
parallel_host_policy const& policy(Kokkos::OpenMP)
{
    return parallel_host;
}
#endif
```